### PR TITLE
Add validation for Dart method overloads

### PR DESCRIPTION
**Pull request by DanielKamkha**
Added DartOverloadsValidator to issue validation warnings for overloaded methods and constructors without @Dart() platform names, as well as for overloaded constructors with multiple @Dart(Default) attributes. In a single set of overloads, one non-renamed method/constructor is fine.

Warnings are issued instead of errors because it's Dart compiler's responsibility to emit these as errors. There are many more such compiler-detectable issues (e.g. language-invalid identifier names). Detecting all of them on Gluecodium side would have added a huge maintenance burden with only a minor upside. Detecting just some of them but not all would make Gluecodium validation inconsistent.

For this particular case, Gluecodium's validation is a supplementary warning system, added on a user's request to partially ease the burden on Dart build system quirks. It is up to the user to treat these warnings as errors (see https://github.com/heremaps/gluecodium/issues/232).

Resolves: #231
Signed-off-by: Daniel Kamkha <daniel.kamkha@here.com>

### CODE REVIEW

File: gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    private fun validateContainer(limeContainer: LimeContainerWithInheritance): Boolean {
+        val allFunctions = limeContainer.functions + limeContainer.inheritedFunctions
+        val constructors = allFunctions.filter { it.isConstructor }
+
+        val overloadedFunctions = (allFunctions - constructors)
+            .groupBy { nameResolver.resolveName(it) }
+            .filter { it.value.size > 1 }
+        overloadedFunctions.forEach { entry ->
+            val pathsString = entry.value.map { it.path.toString() }.sorted().joinToString()
+            logger.warning(
```
Comment (Hsilgos) [inline gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt:54]: Isn't it error?

File: gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        val overloadedConstructors =
+            constructors.groupBy { getConstructorName(it) }.filter { it.value.size > 1 }
+        overloadedConstructors.forEach { entry ->
+            val pathsString = entry.value.map { it.path.toString() }.sorted().joinToString()
+            val warningText = if (entry.key.isEmpty()) {
+                "there is more than one default constructor: $pathsString"
+            } else {
+                "there is more than one constructor with '${entry.key}' name: $pathsString"
+            }
+            logger.warning(entry.value.first(), warningText)
```
Comment (Hsilgos) [inline gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt:69]: the same

File: gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt (hunk 1 above)
Comment (DanielKamkha) [reply gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt:54]: I updated the commit message with an explanation.

File: gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt (hunk 2 above)
Comment (DanielKamkha) [reply gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartOverloadsValidator.kt:69]: I updated the commit message with an explanation.

**Pull request merged**